src/components/BounceRuleDetailed/index.jsx

```diff
@@ -1,61 +1,531 @@
+  isConfirmOpen,
+  isCancelConfirmOpen,
+  handleModalConfirm,
+  handleModalClose,
+  onChangeRule,
+  handleButtonClicked,
+}) => (
+  <div className="detailed-page-container">
+    <Header />
+    <Row>
+      <Column width={6} offset={2}>
+        <Breadcrumb>
+          <Link to="/bounce_rules">Bounce Rules</Link>
+          <Link to={`/bounce_rules/${currentRule.id}`}>{currentRule.id}</Link>
+        </Breadcrumb>
+      </Column>
+    </Row>
+    <Row>
+      <Column width={6} offset={2}>
+        <h1>
+          {"Bounce Rule"} {currentRule.id}
+        </h1>
+      </Column>
+      {isEditClicked ? (
+        <Column className=" sg-right" width={4} offset={8}>
+          <Button
+            onClick={() => handleButtonClicked("cancelClicked")}
+            onKeyDown={() => handleButtonClicked("cancelClicked")}
+            data-test="cancel-button"
+            className="header-button cancel-button"
+            type="secondary"
+          >
+            {"Cancel"}
+          </Button>
+          <Button
+            onClick={() => handleButtonClicked("saveClicked")}
+            onKeyDown={() => handleButtonClicked("saveClicked")}
+            data-test="save-button"
+            className="header-button save-button"
+            type="primary"
+          >
+            {"Save"}
+          </Button>
+        </Column>
+      ) : (
+        <Column
+          className="details-button-column sg-right"
+          width={1}
+          offset={11}
+        >
+          <span>
+            <Button
+              onClick={() => handleButtonClicked("editClicked")}
+              onKeyDown={() => handleButtonClicked("editClicked")}
+              data-test="edit-button"
+              className="header-button edit-button"
+              type="primary"
+            >
+              {"Edit Rule"}
+            </Button>
+          </span>
+        </Column>
+      )}
+      {/* </Column> */}
+    </Row>
+    <Row>
+      <Column width={10} offset={2}>
+        <div>
+          {isEditClicked ? (
+            <DetailContainerEditable
+              data-test="detailed-container-editable"
+              currentRule={currentRule}
+              onChangeRule={onChangeRule}
+              newRule={newRule}
+            />
+          ) : (
+            <DetailContainer
+              data-test="detailed-container"
+              currentRule={currentRule}
+              handleButtonClicked={handleButtonClicked}
+            />
+          )}
         </div>
-        <button type="submit" className="btn btn-primary">
-          Create a Bounce Rule
-        </button>
-      </div>
-      <div
-        style={{
-          border: "1px solid black",
-          padding: "1.5rem",
-          margin: "2rem 0",
-        }}
-      >
-        {currentRule.description}
-      </div>
-      <div>Changelog</div>
-    </div>
+      </Column>
+    </Row>
+    <Row className="changelog-container">
+      <Column width={10} offset={2}>
+        <Changelog
```

Comment:
let's make this file a lot smaller I would break out the SFCs into more folders nested underneath this one like the Changelog related things and say the details edit view or other things so we can have individual stories for those and possibly unit tests and then import them all in this page level container. it'll be easier to maintain and watch for design changes on each component when we have stories for each of them and unit tests for each of those components